dev/AppNotifications/AppNotification.cpp

```diff
@@ -6,8 +6,15 @@ using namespace winrt::Windows::Data::Xml::Dom;
 
 namespace winrt::Microsoft::Windows::AppNotifications::implementation
 {
+    AppNotification::AppNotification()
+    {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_AppNotifications::IsEnabled());
+    }
+
     AppNotification::AppNotification(hstring const& payload)
     {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_AppNotifications::IsEnabled());
```

Comment:
Not needed. See comment above.

Reply:
Actually I see this being put everywhere else? Why not gate all the entire feature via AppManager::Default() instead of these individual checks?

Reply:
I had added this into the ctor of each public runtime class, that's what push seems to be doing. 

I've removed it from all ctor (for toast) and only added one single check in AppNotificationManager::Default().